src/fsharp/QuotationTranslator.fs

```diff
@@ -620,12 +712,46 @@ and private ConvExprCore cenv (env : QuotationTranslationEnv) (expr: Expr) : QP.
         | TOp.RefAddrGet _, _, _ ->
             wfail(Error(FSComp.SR.crefQuotationsCantRequireByref(), m))
 
-        | TOp.TraitCall (_ss), _, _ ->
-            wfail(Error(FSComp.SR.crefQuotationsCantCallTraitMembers(), m))
+        | TOp.TraitCall traitInfo, _, args ->
+            let g = g
+            let inWitnessPassingScope = not env.witnessesInScope.IsEmpty
+            let witnessArgInfo = 
+                if g.generateWitnesses && inWitnessPassingScope then 
+                    match env.witnessesInScope.TryGetValue traitInfo.TraitKey with 
+                    | true, storage -> Some storage
+                    | _ -> None // failwithf "no storage for witness %s found in scope" w.MemberName
```

Comment:
nit: Remove commented code.

If this actually ever returned None, should it actually throw though?

Reply:
Removed the commented code.  The None case does assert below - I think I'd prefer to code this defensively as is (because so many important scenarios still continue to work even if witnesses generation returns None due to a witness not being found correctly in the environment)